src/library.tests/WorkloadInformationProviderTests.cs

```diff
@@ -1,109 +1,261 @@
+
+            ConfigureHeadlessServiceWithIgnorePorts(numAddresses, ignorePorts, servicePorts);
+            var result = await _workloadInformationProvider.GetReachableEndpointsAsync(namespaceName: "", localProcessConfig: null, includeSameNamespaceServices: true, cancellationToken: default(CancellationToken));
+            
+            // Find all ports which were allocated to each service
+            Dictionary<string, List<int>> portsAllocatedToService = new Dictionary<string, List<int>>();
+            foreach (var endpointInfo in result) 
+            {
+                var ports = portsAllocatedToService.GetOrAdd(endpointInfo.DnsName, () => new List<int>());
+                ports.AddRange(endpointInfo.Ports.Select( p => p.RemotePort));               
+            }  
+            
+            // For each service ensure that none of the ignore ports are in the ports allocated and all other servicePorts are allocated to the service
+            Assert.True(CompareLists(ignorePorts.Keys.Select( s => $"host-0.{s}.").ToList(), portsAllocatedToService.Keys.ToList()));
+            
+            foreach(var serviceName in ignorePorts.Keys) 
+            {
+                // Identify expected ports for a service.
+                var ignorePortsList = ignorePorts.GetValueOrDefault(serviceName)?.Split(",").Select(p => int.Parse(p)).ToList() ?? new List<int>();
+                var expectedPortsForService = servicePorts.GetValueOrDefault(serviceName)?.Where( p => !ignorePortsList.Contains(p)) ?? new List<int>();
+
+                // Check if the allocated ports (from result) are same as the expected ports for the service.
+                Assert.True(CompareLists(expectedPortsForService.ToList(), portsAllocatedToService.GetValueOrDefault($"host-0.{serviceName}.")));
+            }        
+        }
+
+        [Theory]
+        [InlineData(1)]
+        public async void GetReachableServicesAsync_PortsToIgnoreIncorrectFormat_HeadlessService(int numAddresses) {
+            Dictionary<string, string> ignorePorts = new Dictionary<string, string>();
+            ignorePorts.Add("ServiceA", "19o, abc");
+            ConfigureHeadlessServiceWithIgnorePorts(numAddresses, ignorePorts);
+            await Assert.ThrowsAsync<UserVisibleException>(() => _workloadInformationProvider.GetReachableEndpointsAsync(namespaceName: "", localProcessConfig: null, includeSameNamespaceServices: true, cancellationToken: default(CancellationToken)));
+        }
+
+        // Comparator when the list elements are unique.
+        private static bool CompareLists<T>(List<T> list1, List<T> list2) 
+        {
+            if (list1.Count != list2.Count)
+            {
+                return false;
+            }
+            return list1.All( item => list2.Contains(item));
+        }
+
+        private void ConfigureHeadlessServiceWithIgnorePorts(int numAddresses, Dictionary<string, string> ignorePorts, Dictionary<string, List<int>> servicePorts = null) 
```

Comment:
I think we should rename this function to ConfigureServiceWithIgnorePorts and ensure we add both types of services otherwise as it is now only half of the code your added is being tested (aka only the headlessservice code path is currently executed).

Reply:
you will want to also rename the UTs to not mention they are only for headless service